examples/kafka/demos/demo.sh

```diff
@@ -0,0 +1,151 @@
+### Use Pulsar client with TLS endpoint in Pulsar with token auth:
+bin/pulsar-perf produce -r 1000 --size 1024 --auth_plugin org.apache.pulsar.client.impl.auth.AuthenticationToken --auth-params file:///pulsar/token-superuser-stripped.jwt --service-url pulsar+ssl://pulsar-proxy.pulsar.svc.cluster.local:6651/ persistent://public/default/test
+
+## OIDC Setup:
+cat > /pulsar/conf/creds.json
+{"client_id":"0oa7ypwvxnvo9xnDd5d7","client_secret":"CL08ZNhF91fsCUm7rtYqHs-XUak5H7gLY01tF2bP","grant_type": "client_credentials"}
+
+### Use Pulsar client with non-TLS endpoint in Pulsar with OIDC:
+bin/pulsar-perf produce -r 1000 --size 1024 --auth-plugin "org.apache.pulsar.client.impl.auth.oauth2.AuthenticationOAuth2" --auth-params '{"privateKey":"/pulsar/conf/creds.json","issuerUrl":"https://dev-42506116.okta.com/oauth2/aus3thh6rqs3FU45X697","scope":"pulsar_client_m2m","audience":"api://pulsarClient"}' --service-url pulsar://pulsar-proxy.pulsar.svc.cluster.local:6650/ persistent://public/default/test
+### Use Pulsar client with TLS endpoint in Pulsar with OIDC:
+bin/pulsar-perf produce -r 1000 --size 1024 --auth-plugin "org.apache.pulsar.client.impl.auth.oauth2.AuthenticationOAuth2" --auth-params '{"privateKey":"/pulsar/conf/creds.json","issuerUrl":"https://dev-42506116.okta.com/oauth2/aus3thh6rqs3FU45X697","scope":"pulsar_client_m2m","audience":"api://pulsarClient"}' --service-url pulsar+ssl://pulsar-proxy.pulsar.svc.cluster.local:6651/ persistent://public/default/test
+
+
+# Deploy credentials obtained from Okta (https://www.youtube.com/watch?v=UQBrecHOXxU&ab_channel=DataStaxDevelopers) or other provider.
+# Note: The Kafka endpoints occasionally change as they lifecycle versions of Kafka releases. If you have an error when unpacking
+#     the tarball, you might be curling a 404, so be sure to check that the URL hasn't changed.
+mkdir /pulsar/kafka && cd /pulsar/kafka
+curl -LOs https://downloads.apache.org/kafka/3.3.2/kafka_2.12-3.3.2.tgz
+tar -zxvf /pulsar/kafka/kafka_2.12-3.3.2.tgz
+cd /pulsar/kafka/kafka_2.12-3.3.2/libs
+curl -LOs https://github.com/datastax/starlight-for-kafka/releases/download/v2.10.3.0/oauth-client-2.10.3.0.jar
+cd ..
+
+########
+# To test OpenID/OAuth2 on non-TLS endpoint:
+# Make sure that the oauth parameters below map to your actual configs in the helm values file!
+cat > /pulsar/kafka/kafka_2.12-3.3.2/config/producer.properties
+bootstrap.servers=pulsar-proxy:9092
+compression.type=none
+sasl.login.callback.handler.class=com.datastax.oss.kafka.oauth.OauthLoginCallbackHandler
+security.protocol=SASL_PLAINTEXT
+sasl.mechanism=OAUTHBEARER
+sasl.jaas.config=org.apache.kafka.common.security.oauthbearer.OAuthBearerLoginModule \
+   required oauth.issuer.url="https://dev-42506116.okta.com/oauth2/aus7ypk6sjvgF4l615d7"\
```

Comment:
can we create a variable in the beginning of the .sh file with `dev-42506116.okta.com` ?